coverage/coverage.go

```diff
@@ -0,0 +1,214 @@
+package coverage
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+)
+
+type Model struct {
+	Bool                    *map[string]bool   `json:"Bool,omitempty"`
+	Discriminator           *string            `json:"Discriminator,omitempty"`
+	Enum                    *map[string]bool   `json:"Enum,omitempty"`
+	Format                  *string            `json:"Format,omitempty"`
+	Identifier              string             `json:"Identifier,omitempty"`
+	IsAnyCovered            bool               `json:"IsAnyCovered"`
+	IsFullyCovered          bool               `json:"IsFullyCovered,omitempty"`
+	HasAdditionalProperties bool               `json:"HasAdditionalProperties,omitempty"`
+	CoveredCount            int                `json:"CoveredCount,omitempty"`
+	TotalCount              int                `json:"TotalCount,omitempty"`
+	EnumCoveredCount        int                `json:"EnumCoveredCount,omitempty"`
+	EnumTotalCount          int                `json:"EnumTotalCount,omitempty"`
+	BoolCoveredCount        int                `json:"BoolCoveredCount,omitempty"`
+	IsReadOnly              bool               `json:"IsReadOnly,omitempty"`
+	IsRequired              bool               `json:"IsRequired,omitempty"`
+	Item                    *Model             `json:"Item,omitempty"`
+	Properties              *map[string]*Model `json:"Properties,omitempty"`
+	SourceFile              string             `json:"SourceFile,omitempty"`
+	Type                    *string            `json:"Type,omitempty"`
+	Variants                *map[string]*Model `json:"Variants,omitempty"`
+}
+
+func (m *Model) MarkCovered(root interface{}) {
+	if root == nil || m == nil || m.IsReadOnly {
+		return
+	}
+
+	m.IsAnyCovered = true
+
+	// https://pkg.go.dev/encoding/json#Unmarshal
+	switch value := root.(type) {
```

Comment:
>switch [](http://example.com/codeflow?start=1&length=6)

not have `integer` ?

Reply:
From https://pkg.go.dev/encoding/json#Unmarshal, only below 6 is possible:
>bool, for JSON booleans
float64, for JSON numbers
string, for JSON strings
[]interface{}, for JSON arrays
map[string]interface{}, for JSON objects
nil for JSON null